### PR TITLE
Add POC grouped arrange

### PR DESCRIPTION
**Pull request by cigrainger**
This is a POC (that works!) showing how we can leverage groups + apply to do more efficient grouped functions. Other verbs (e.g. filter) will require expressions. But this one works!

To do before merge:
- [x] add grouped tests
- [ ] ~benchmark~ I'm going to leave benchmarking for now and work it into a 0.2.0 blog post

### CODE REVIEW

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
Btw, I just realized that we should not make a similar change to  `mutate` yet because we will change mutate to rely on expressions, and moving it to Rust will likely make that particular process harder. So we should hold on a little bit. Distinct should be doable though!

**Review by philss — COMMENTED** [review]
Cool! I liked the idea! It would make thinks clearer in the Elixir side!

**Comment by cigrainger** [comment]
> Btw, I just realized that we should not make a similar change to mutate yet because we will change mutate to rely on expressions, and moving it to Rust will likely make that particular process harder. So we should hold on a little bit. Distinct should be doable though!

Oh yeah distinct is doable! Yep we need expressions for most of them. Let me roll distinct into this PR.

File: lib/explorer/polars_backend/data_frame.ex [hunk 1]
```diff
@@ -316,23 +316,21 @@ defmodule Explorer.PolarsBackend.DataFrame do
...
-
-  def arrange(%DataFrame{groups: [_ | _]} = df, columns) do
-    apply_on_groups(df, df, fn group -> ungrouped_arrange(group, columns) end)
-  end
+  def arrange(%DataFrame{groups: groups} = df, columns) do
+    groups =
+      case groups do
+        [] -> nil
+        groups -> groups
+      end
```
[the remark this answers is not in the document]
Comment (cigrainger) [reply lib/explorer/polars_backend/data_frame.ex:324]: Yeah I did this one 'backwards' in that I wrote the Rust code then worked back out to the Elixir code. I'll switch it around to use `is_empty`.

**Comment by cigrainger** [comment]
Hmmmm there's a flaky test. Passing most of the time but not always.

**Comment by cigrainger** [comment]
Okay I've switched from `DataFrame::groupby` to `DataFrame::groupby_stable` so we're getting predictable outputs. I believe this is in line with dplyr behaviour as well.

**Comment by philss** [comment]
> Okay I've switched from DataFrame::groupby to DataFrame::groupby_stable so we're getting predictable outputs. I believe this is in line with dplyr behaviour as well.

Cool! I will copy this to the change needed in #254

**Review by philss — APPROVED** [review]
:shipit:

**Pull request merged**